spyder/api/plugin_registration/registry.py

```diff
@@ -457,6 +458,9 @@ def delete_plugin(self, plugin_name: str, teardown: bool = True) -> bool:
         teardown: bool
             True if the teardown notification to other plugins should be sent
             when deleting the plugin, False otherwise.
+        check_can_delete: bool
+            True if the plugin should validate if can be closed in the moment,
+            False otherwise.
```

Comment:
```suggestion
            True if the plugin should validate if it can be closed when this
            method is called, False otherwise.
```

I'm not so sure if this exactly what you mean here, but it seems like it. If not, please edit my suggestion.